component/http/middleware.go

```diff
@@ -133,17 +139,74 @@ func NewAuthMiddleware(auth auth.Authenticator) MiddlewareFunc {
 }
 
 // NewLoggingTracingMiddleware creates a MiddlewareFunc that continues a tracing span and finishes it.
-// It also logs the HTTP request on debug logging level
+// It uses Jaeger and OpenTracing and will also log the HTTP request on debug level if configured so.
 func NewLoggingTracingMiddleware(path string, statusCodeLogger statusCodeLoggerHandler) MiddlewareFunc {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			corID := getOrSetCorrelationID(r.Header)
 			sp, r := span(path, corID, r)
-			lw := newResponseWriter(w)
+			lw := newResponseWriter(w, true)
 			next.ServeHTTP(lw, r)
-			finishSpan(sp, lw.Status(), lw.payload)
+			finishSpan(sp, lw.Status(), &lw.responsePayload)
 			logRequestResponse(corID, lw, r)
-			statusCodeErrorLogging(r.Context(), statusCodeLogger, lw.Status(), lw.payload, path)
+			if log.Enabled(log.ErrorLevel) && statusCodeLogger.shouldLog(lw.status) {
+				log.FromContext(r.Context()).Errorf("%s %d error: %v", path, lw.status, string(lw.responsePayload.Bytes()))
+			}
+		})
+	}
+}
+
+var (
```

Comment:
I can move this block to the top of the file (after constants) if we prefer that style.

Reply:
yes, this makes more sense.